File: recrutplus-backend/src/main/java/com/recrutplus/recrutplus/model/Application.java
package com.recrutplus.recrutplus.model;

import com.recrutplus.recrutplus.model.enums.ApplicationStatus;
import jakarta.persistence.*;
import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;

@Entity
@Table(name = "application")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Application {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String firstname;

    @Column(nullable = false)
    private String lastname;

    @Column(nullable = false)
    @Email
    private String email;

    @Column(nullable = false)
    @Size(min = 10)
    private String phone;

    @Column
    private String cvPath;

    @Column(columnDefinition = "TEXT")
    private String coverLetter;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private ApplicationStatus status;

    @Column(nullable = false, updatable = false)
    private LocalDateTime applicationDate;

    private LocalDateTime processedAt;

    @Column(columnDefinition = "TEXT")
    private String comment;

    private LocalDateTime updatedAt;

    @ManyToOne
    @JoinColumn(name = "job_offer_id", nullable = false)
    private JobOffer jobOffer;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @OneToMany(mappedBy = "application", cascade = CascadeType.ALL)
    private List<Interview> interviews;
}

File: recrutplus-frontend/src/types/Application.ts
import type { Address } from './Address';
import type { JobOffer } from './JobOffer';
import type { User } from './User';

export enum ApplicationStatus {
  EN_ATTENTE = 'EN_ATTENTE',
  EN_COURS = 'EN_COURS', 
  ACCEPTE_ENTRETIEN = 'ACCEPTE_ENTRETIEN',
  ENTRETIEN_TERMINE = 'ENTRETIEN_TERMINE',
  REFUSE = 'REFUSE',
  EMBAUCHE = 'EMBAUCHE',
  REFUSE_APRES_ENTRETIEN = 'REFUSE_APRES_ENTRETIEN'
}

export interface Application {
  id: number;
  firstname: string;
  lastname: string;
  email: string;
  phone: string;
  birthdate?: string;
  address?: Address;
  cvPath?: string;
  coverLetter: string;
  status: ApplicationStatus;
  applicationDate: string;
  processedAt?: string;
  comment?: string;
  updatedAt?: string;
  jobOffer: JobOffer;
  user?: User;
}

export interface CreateApplicationDTO {
  jobOfferId: number;
  firstname: string;
  lastname: string;
  email: string;
  phone: string;
  birthdate?: string;
  address?: Address;
  coverLetter: string;
}

export interface UpdateApplicationDTO {
  firstname?: string;
  lastname?: string;
  email?: string;
  phone?: string;
  coverLetter?: string;
  status?: ApplicationStatus;
  comment?: string;
}

export interface ProcessApplicationDTO {
  status: ApplicationStatus;
  comment?: string;
}